Assets/Scripts/Traps/PressurePlate.cs

```diff
@@ -0,0 +1,28 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PressurePlate : MonoBehaviour, Interact
+{
+
+
+
+
+
+    private void OnTriggerEnter(Collider other)
+    {
+        Debug.Log("detected");
+
+        if (other.gameObject.GetComponent<IEntityHealth>() != null)
+        {
+            Debug.Log("Pressure detected");
+            Interact();
+        }
+    }
+
+
+    public void Interact()
+    {
+        GetComponentInParent<ProyectileTrap>().CheckShoot();
```

Comment:
I made `PressurePlate` as a child of `ProjectileTrap` so it could be easier to setup on scene, but I can call it with a reference using `[SerializeField]` if you find it more appropriate.